Assets/Scripts/Framework/GeoLocation/RouteHighlighter.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace Framework.GeoLocation
+{
+    [RequireComponent(typeof(LineRenderer))]
+    public sealed class RouteHighlighter : MonoBehaviour
+    {
+        private const float INVOKE_DELAY = 0.1f;
+
+        [SerializeField] private UnityEvent withinRange = new UnityEvent();
```

Comment:
```suggestion
        [SerializeField] private UnityEvent onWithinRange = new UnityEvent();
        [SerializeField] private UnityEvent onRouteDone = new UnityEvent();
```
Renaming + een nieuwe event.